src/controllers/dynakube/activegate/internal/proxy/reconciler_test.go

```
@@ -0,0 +1,177 @@
+		r := newTestReconciler(fake.NewClientBuilder().Build())
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{Value: ""}
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Equal(t, []byte(nil), proxySecret.Data[proxyPasswordField])
+		assert.Equal(t, []byte(nil), proxySecret.Data[proxyPortField])
+		assert.Equal(t, []byte(nil), proxySecret.Data[proxyHostField])
+		assert.Equal(t, []byte(nil), proxySecret.Data[proxyUsernameField])
+		assert.True(t, update)
+		assert.NoError(t, err)
+	})
+}
+
+func TestReconcileProxyValueFrom(t *testing.T) {
+	var proxyUrl = buildProxyUrl(proxyUsername, proxyPassword, proxyHost, proxyPort)
+	var testClient = fake.NewClientBuilder().WithObjects(createProxySecret(proxyUrl)).Build()
+	r := newTestReconciler(testClient)
+
+	t.Run(`reconcile proxy ValueFrom`, func(t *testing.T) {
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{ValueFrom: customProxySecret}
+		update, err := r.Reconcile()
+
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
+
+		assert.Equal(t, []byte(proxyPassword), proxySecret.Data[proxyPasswordField])
+		assert.Equal(t, []byte(proxyPort), proxySecret.Data[proxyPortField])
+		assert.Equal(t, []byte(proxyHost), proxySecret.Data[proxyHostField])
+		assert.Equal(t, []byte(proxyUsername), proxySecret.Data[proxyUsernameField])
+		assert.True(t, update)
+		assert.NoError(t, err)
+	})
+	t.Run(`Change of Proxy ValueFrom to Value`, func(t *testing.T) {
+		r.dynakube.Spec.Proxy = &dynatracev1beta1.DynaKubeProxy{ValueFrom: customProxySecret}
+		update, err := r.Reconcile()
+		var proxySecret corev1.Secret
+		r.client.Get(context.TODO(), client.ObjectKey{Name: capability.BuildProxySecretName(), Namespace: testNamespace}, &proxySecret)
```

same as bevore, do not implicitly ignore an error